pkg/gadgets/trace/oomkill/tracer/tracer_test.go

```diff
@@ -0,0 +1,195 @@
+	// allocate the memory until it gets killed.
+	err = os.Mkdir(cgroupPath, 0o755)
+	require.NoError(t, err, "creating cgroup path %s", cgroupPath)
+	defer syscall.Rmdir(cgroupPath)
+
+	memoryMaxPath := fmt.Sprintf("%s/memory.max", cgroupPath)
+	file, err := os.OpenFile(memoryMaxPath, os.O_WRONLY, 0)
+	require.NoError(t, err, "opening %s", memoryMaxPath)
+
+	_, err = file.Write([]byte(fmt.Sprintf("%d\n", memoryLimit)))
+	file.Close()
+	require.NoError(t, err, "writing %d to %s", memoryLimit, memoryMaxPath)
+
+	events := make(chan *types.Event)
+	eventCallback := func(event *types.Event) {
+		// normalize
+		event.Timestamp = 0
+
+		events <- event
+	}
+
+	sock, err := unix.Socketpair(unix.AF_UNIX, unix.SOCK_DGRAM, 0)
+	require.NoError(t, err, "creating socketpair")
+
+	runner := utilstest.NewRunnerWithTest(t, nil)
+
+	createTracer(t, &tracer.Config{
+		MountnsMap: utilstest.CreateMntNsFilterMap(t, runner.Info.MountNsID),
+	}, eventCallback)
+
+	createTracer(t, &tracer.Config{}, eventCallback)
+
+	utilstest.RunWithRunner(t, runner, func() error {
+		// Use clone() to make it more portable, at least for amd64 and arm64.
+		childPid, _, errno := syscall.Syscall6(syscall.SYS_CLONE, uintptr(syscall.SIGCHLD), 0, 0, 0, 0, 0)
+		require.Equal(t, errno, syscall.Errno(0x0), "spawning child process")
+
+		if childPid == 0 {
+			unix.Recvmsg(sock[1], []byte{}, []byte{}, 0)
+
+			// The child is now part of the memory cgroup, let's allocate some memory.
+			// The memory will be automatically touched by golang as it will initialize
+			// to 0, so no lazy allocation there.
+			// As a result, the child should be OOM killed.
+			// NOLINTNEXTLINE
+			_ = make([]byte, memoryLimit*100)
+
+			// This code should never get reached.
+			os.Exit(1)
+		}
+
+		procsPath := fmt.Sprintf("%s/cgroup.procs", cgroupPath)
+		procsFile, err := os.OpenFile(procsPath, os.O_WRONLY, 0)
+		require.NoError(t, err, "opening %s", procsPath)
+
+		_, err = procsFile.Write([]byte(fmt.Sprintf("%d\n", childPid)))
+		procsFile.Close()
+		require.NoError(t, err, "writing %d to %s", childPid, procsPath)
+
+		proc, err := os.FindProcess(int(childPid))
+		require.NoError(t, err, "no process with PID %d", childPid)
+
+		// Unblock the child.
+		unix.Sendmsg(sock[0], []byte{}, []byte{}, nil, 0)
+
+		done := make(chan error)
+		go func() {
+			_, err := proc.Wait()
+			done <- err
+		}()
+
+		select {
+		case err := <-done:
+			require.NoError(t, err, "waiting child with PID %d", childPid)
+		case <-time.After(10 * time.Second):
+			t.Fatalf("waiting child with PID %d: time out", childPid)
+		}
+
+		select {
+		case event := <-events:
+			killedPid := event.KilledPid
+			require.Equal(t, killedPid, uint32(childPid), "killedPid is %d, expected %d", killedPid, childPid)
+
+			killedComm := event.KilledComm
+			childComm := path.Base(os.Args[0])
+			require.Equal(t, killedComm, childComm, "killedComm is %s, expected %s", killedComm, childComm)
```

Comment:
We can't test any of the `Triggered` fields because it can be any process, right?